kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -61,6 +62,33 @@ object StreamCoreExtensions:
                             _ <- emitMaybeChunksFromChannel(channel)
                         yield ()
 
+        /** Creates a stream from an iterator.
+          *
+          * @param v
+          *   Iterator to create a stream from
+          * @param bufferSize
+          *   Size of the buffer that the iterator will write to and the stream will read from
+          */
+        def fromIterator[V, S](v: => Iterator[V], bufferSize: Int = Stream.DefaultChunkSize)(using
+            tag: Tag[Emit[Chunk[V]]],
+            frame: Frame
+        ): Stream[V, IO] =
+            val builder = ChunkBuilder.init[V]
+
+            def nextElement(it: Iterator[V])(using Frame): Maybe[Chunk[V]] < IO =
+                val size = bufferSize max 1
+                IO.ensure(builder.clear()):
+                    var count = 0
+                    while count < size && it.hasNext do
+                        builder.addOne(it.next())
+                        count += 1
+
+                    Maybe.when(count > 0)(builder.result())
+            end nextElement
+
+            Stream.repeatPresent(nextElement(v))
+        end fromIterator
```

Comment:
@hearnadam following https://github.com/getkyo/kyo/pull/1222#discussion_r2118238010
It is done like that.